RecoEgamma/EgammaElectronProducers/plugins/TrackingRegionsFromSuperClustersProducer.h

```diff
@@ -239,14 +258,21 @@ getVtxPos(const edm::Event& iEvent,double& deltaZVertex)const
   //we fall back to beamspot mode
   auto beamSpotHandle = getHandle(iEvent,beamSpotToken_);
   const reco::BeamSpot::Point& bsPos = beamSpotHandle->position();
-  //SH: this is what SeedFilter did, no idea what its trying to achieve....
-  const double sigmaZ = beamSpotHandle->sigmaZ();
-  const double sigmaZ0Error = beamSpotHandle->sigmaZ0Error();
-  deltaZVertex = 3*std::sqrt(sigmaZ*sigmaZ+sigmaZ0Error*sigmaZ0Error);
-  //  std::cout <<"z "<<bsPos.z()<<" deltaZ "<<deltaZVertex<<std::endl;
-  //deltaZVertex = 30.;
-  return GlobalPoint(bsPos.x(),bsPos.y(),bsPos.z());
-  //return GlobalPoint(bsPos.x(),bsPos.y(),0);
+  
+  if(useZInBeamspot_){
```

Comment:
I hate silent fall backs and not sure how this one slipped past me. An exception will now be thrown if the vertex collection is not there and we were expecting it to be. Thanks for the spot